server/src/main/java/org/elasticsearch/action/admin/indices/recovery/RecoveryResponse.java

```diff
@@ -37,7 +36,7 @@
 /**
  * Information regarding the recovery state of indices and their associated shards.
  */
-public class RecoveryResponse extends BroadcastResponse implements ToXContentFragment {
+public class RecoveryResponse extends BroadcastResponse {
 
     private boolean detailed = false;
```

Comment:
I just figured this flag is never read, we can rather remove it and simplify the rest action. We should also remove it from the request as it does nothing, and adjust the docs. The correct parameter is `details` and not `detailed`, but we don't accept it anymore since we introduced params validation a while ago. The flag doesn't need to be serialized either, as it only affects the xcontent output, is is simply a REST param passed through to the response like we do in other places. Would you like to open a follow-up PR to fix this?

Reply:
Yes, cool ! of course :) I'll look at it after finishing these PRs, and it's like this it changes a little whlie working on Add Broadcast API :smile:

Reply:
I opened #28910 to keep track of this.

Reply:
what do you think of removing the flag already from the response given that it does nothing? I think it would simplify the rest action